icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/PersonNameFormatterTest.java

```diff
@@ -507,4 +507,35 @@ public void TestCapitalization() {
             assertEquals("Wrong result", expectedResult, actualResult);
         }
     }
+
+    @Test
+    public void TestLocaleDerivation() {
+        // Test for https://unicode-org.atlassian.net/browse/ICU-22379, which implements the algorithm
+        // described in https://unicode-org.atlassian.net/browse/CLDR-16623.
+        executeTestCases(new NameAndTestCases[]{
+            // If we have a name that's tagged as Japanese, but contains Latin characters, and we're using
+            // a Japanese formatter, we actually use the English formatter to format it, but because the name is
+            // tagged as Japanese, we still use Japanese field order
+            new NameAndTestCases("given=Richard,surname=Gillam,locale=ja_AQ", new String[][]{
```

Comment:
Add some cases with ja_Hani and ja_Jpan, ko_Hang, zh_Hant, and zh_Hans, just to make sure those work.

Reply:
Added a few new ones for Japanese to get at this issue.  I didn't add any for Korean or Chinese, on the assumption that if Japanese works, they will too, but if you really want them, I'll try to figure something out.